Automattic/jetpack
# Build: Server-side Render UpgradeNudge for use in PHP

Fixes #13040.

This PR adds infrastructure to server-side render individual React components during build so they can be used in PHP. The idea is that props can be passed to the component using a simplistic templating language on the server side. The benefit is that we'll be able to re-use markup and styling and don't have to duplicate code in PHP. 

This is inspired by [prior art in `static.jsx`](https://github.com/Automattic/jetpack/blob/f25e25705c67146317c1cb7334856352bd47b44b/webpack.config.js#L43-L76) -- see e.g.#12381 -- but hopes to apply the same principle in a cleaner, more granular way (component level).

It also adds functionality to fetch plans data from the relevant WP.com endpoint.

#### Changes proposed in this Pull Request:

Build: Server-side Render UpgradeNudge for use in PHP

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

Adds feature to existing part.

#### Testing instructions:

Create a new jurassic.ninja site, connect to WP.com, selecting a free plan.
Start a new post and insert the Simple Payments block. Fill in some information and publish the post.
Verify that in the frontend, you see a warning that looks like in the editor.

![image](https://user-images.githubusercontent.com/96308/61873412-c8341880-af07-11e9-9221-0c7e633c5120.png)

#### TODO

- [x] Styling currently ends up in `static.css`. We'd like a separate style file. Do we even need the rest in `static.css`? Can we drop that?
- [x] Don't `import React from 'react'`, use `@wordpress/element` instead
- [x] Add error handling for plans fetching

#### Follow-up

Minify styling

/cc @sirreal
